### ISSUE TITLE
Purpose and explanation of opt.scale1 and opt.scale2

### ISSUE THREAD

**Issue by erhuliu** [Hzzone/TCL#5]
````
estimated_noise_ratio = (confidence > 0.5).float().mean().item()
        if opt.scale1 is None:
            self.tcl.module.scale1 = estimated_noise_ratio
        if opt.scale2 is None:
            self.tcl.module.scale2 = estimated_noise_ratio
````

`targets_corrected1 = comb(q_prob2, targets_onehot_noise, confidences * self.scale1)`
`targets_corrected2 = comb(q_prob1, targets_onehot_noise, confidences * self.scale1)`
`targets_mix_corrected = comb((q_prob1 + q_prob2) * 0.5, targets_onehot_noise, confidences * self.scale2)`

What is the purpose of multiplying by this scale? Can you explain it?

**Comment by Hzzone** [comment]
The two scales are the hyper-parameters to rescale the global confidence of mixup and strong augmentation between estimated targets and gt labels, dependent on the dataset and noise ratios. In low-level noise, the labels are more confident and so we could set the scale to 1, versus for high-level noise.

**Comment by erhuliu** [comment]
Thanks for the reply and one more question

`context_assignments_logits = features.mm(centers.T) / opt.temp`

Why is this divided by the hyperparameter temp, instead of the variance in the paper?

**Comment by Hzzone** [comment]
This setting follows [Prototypical contrastive learning of unsupervised representations], but we found no difference in performance.
